tests/api_test.py

```diff
@@ -219,6 +255,74 @@ def test_get_newdle(flask_client, dummy_newdle):
     }
 
 
+@pytest.mark.usefixtures('db_session')
+def test_update_invalid_newdle(flask_client, dummy_uid):
+    resp = flask_client.patch(
+        url_for('api.update_newdle', code='xxx'),
+        json={'title': 'foo'},
+        **make_test_auth(dummy_uid),
+    )
+    assert resp.status_code == 404
+    assert resp.json == {'error': 'Specified newdle does not exist'}
+
+
+@pytest.mark.usefixtures('db_session')
+def test_update_newdle_unauthorized(flask_client, dummy_newdle):
+    resp = flask_client.patch(
+        url_for('api.update_newdle', code='dummy'),
+        json={'title': 'foo'},
+        **make_test_auth('someone'),
+    )
+    assert resp.status_code == Forbidden.code
+    assert resp.json == {'error': Forbidden.description}
+
+
+@pytest.mark.usefixtures('dummy_newdle')
+def test_update_newdle(flask_client, dummy_newdle, dummy_uid):
+    final_dt = '2019-09-12T13:30'
+    dummy_newdle_json = {
+        'code': 'dummy',
+        'creator_name': 'Dummy',
+        'duration': 60,
+        'final_dt': None,
+        'id': dummy_newdle.id,
+        'timeslots': [
+            '2019-09-11T13:00',
+            '2019-09-11T14:00',
+            '2019-09-12T13:00',
+            '2019-09-12T13:30',
+        ],
+        'timezone': 'Europe/Zurich',
+        'title': 'Test event',
+        'url': 'http://flask.test/newdle/dummy',
+    }
+    json = {
+        'code': 'xxx',
+        'creator_name': 'someone',
+        'duration': 120,
+        'final_dt': final_dt,
+        'id': 10,
+        'timeslots': [
+            '2019-08-11T13:00',
+            '2019-08-11T14:00',
+            '2019-08-12T13:00',
+            '2019-08-12T13:30',
+        ],
+        'timezone': 'Europe/Paris',
+        'title': 'Test event1',
+        'url': 'http://flask.test/newdle/dummy1',
+    }
+    resp = flask_client.patch(
+        url_for('api.update_newdle', code='dummy'),
+        **make_test_auth(dummy_uid),
+        json=json,
+    )
+    dummy_newdle_json['final_dt'] = final_dt
```

Comment:
Why not just set it directly when defining the expected answer above? You never use `dummy_newdle_json` before in any way where you would need access to the original data.

Reply:
I thought it would make it clearer that we only expect 'final_dt' property to be updated even if all the properties were updated when submitted. Happy to change it if necessary !

Reply:
I'd change it ;)

Another option would be to get the newdle json before the change and after, pop the `final_dt` from both, and then assert that they are identical (and that the popped `final_dt` matches what you PATCHed)